lib.rs

```diff
@@ -313,6 +313,32 @@ impl<A: Array> SmallVec<A> {
             self.set_len(len + 1);
         }
     }
+
+    pub fn insert_many<I: IntoIterator<Item=A::Item>>(&mut self, index: usize, iterable: I) {
+        let iter = iterable.into_iter();
+        let (lower_size_bound, _) = iter.size_hint();
+        self.reserve(lower_size_bound);
+
+        unsafe {
+            let ptr = self.as_mut_ptr().offset(index as isize);
```

Comment:
All of these casts to isize when offsetting are unsafe, since they could become negative numbers and cause us to write outside of the bounds of the array. We should either use `to_isize.unwrap()` and panic if that occurs, or use a conversion strategy that yields a value that will gives us worse performance but correct behaviour.